docs/DeveloperGuide.md

```diff
@@ -1,29 +1,342 @@
+- The respective classes (eg `ParseAdd`, `ParseSort`) will be called.
+- For commands that requires an additional input after the command work (excludes commands like `list`, `help` etc), the
+  `ParseIndividualValue` class will be called to parse all the input fields by the user.
+
+The following shows the UML diagram used for the parser component implemented in MyLedger.
+
+<p align="center">
+    <img src="team/images/parserOverview.png">
+    <br/>
+    <i>Figure 2: UML diagram for the parser component</i>
+</p>
+
+It must be noted that not all the existing parser commands are included in this sequence diagram for parsing, namely
+the mark, unmark and edit commands. This is because they have a similar sequence diagram as the functions parseAdd and 
+parseLendBorrow. The only difference is the condition, with the loop happening one, one and four time(s) respectively. 
+
+
+### 3.3. Expenditure Categories
+The **[API](https://github.com/AY2223S2-CS2113-T14-3/tp/blob/master/src/main/java/seedu/expenditure/Expenditure.java)** of this component is specified in the super abstract class `Expenditure.java` and its sub-classes. Its sub-classes represent the different expenditure categories. When users create a new expenditure record, one of these different expenditure categories are instantiated. After which, the expenditure is added to the expenditure list.
+
+The **[API](https://github.com/AY2223S2-CS2113-T14-3/tp/blob/master/src/main/java/seedu/expenditure/ExpenditureList.java)** of the expenditure list is specified in the `ExpenditureList.java` class.
+
+The `ExpenditureList` class description is as follows:
+- A representation of a list of expenditures. It is an `ArrayList<Expenditure>` container
+- The list is instantiated at the start of the program and stores expenditures of type `Expenditure`.
+
+As for the expenditure types, their fields are as shown below.
+
+`Expenditure`:
+- Fields: `date`, `amount`, `description`
+
+`AcademicExpenditure`**`: 
+- Fields: `date`, `amount`, `description`
+
+`AccommodationExpenditure`:
+- Fields: `date`, `amount`, `description`, `isPaid`
+
+`BorrowExpenditure`:
+- Fields: `date`, `borrowerName`, `amount`, `deadline`, `description`
+
+`EntertainmentExpenditure`:
+- Fields: `date`, `amount`, `description`
+
+`FoodExpenditure`: 
+- Fields: `date`, `amount`, `description`
+
+`LendExpenditure`:
+- Fields: `date`, `borrowerName`, `amount`, `deadline`, `description`
+
+`OtherExpenditure`:
+- Fields: `date`, `amount`, `description`
+
+`TransportExpenditure`:
+- Fields: `date`, `amount`, `description`
+
+`TuitionExpenditure`:
+- Fields: `date`, `amount`, `description`, `isPaid`
+
+The following shows the UML diagram used for the Expenditure Categories component implemented in MyLedger.
+
+<p align="center">
+    <img src="team/images/UMLClassDiagramExpenditure.png" width="80%">
+    <br/>
+    <i>Figure 3: UML diagram for the Expenditure Categories component</i>
+</p>
+
```

Comment:
Very clear and easy to read UML Diagram provided. Also, good explanation of the diagram.